partitionmanager/table_append_partition.py

```diff
@@ -420,12 +402,132 @@ def _plan_partition_changes(
             lambda f: f.timestamp() < evaluation_time, filled_partitions
         )
         rate_relevant_partitions = list(filled_partitions) + [
-            partitionmanager.types.InstantPartition(evaluation_time, current_position)
+            partitionmanager.types.InstantPartition(
+                "p_current_pos", evaluation_time, current_position
+            )
         ]
 
+    return rate_relevant_partitions
+
+
+def _get_rate_partitions_with_queried_timestamps(
+    database, table, partition_list, current_position, evaluation_time, active_partition
+):
+    """ Return a list of PositionPartitions for use in rate calculations.
+
+    The partitions' timestamps are explicitly queried.
+    """
+    log = logging.getLogger(
+        f"_get_rate_partitions_with_queried_timestamps:{table.name}"
+    )
+
+    if not table.has_date_query:
+        raise ValueError("Table has no defined date query")
+
+    instant_partitions = list()
+
+    for partition in partition_list:
+        if len(partition.position) != 1:
+            raise ValueError(
+                "This method is only valid for single-column partitions right now"
+            )
+        arg = partition.position.as_sql_input()[0]
+
+        sql_select_cmd = table.earliest_utc_timestamp_query.get_statement_with_argument(
+            arg
+        )
+        log.debug(
+            "Executing %s to derive partition %s at position %s",
```

Comment:
Should this be an f string to keep style with the rest of the function?

Reply:
JC stated that using %s, as recommended by pylint and flake8, should use the parameterized version he has here of %s which is lazily evaluated rather than f strings which must return even if the DEBUG log level isn't used.

Reply:
Citation: https://pycodequ.al/docs/pylint-messages/w1203-logging-fstring-interpolation.html?highlight=log